django/contrib/admin/options.py

```diff
@@ -1814,7 +1814,12 @@ def _get_obj_does_not_exist_redirect(self, request, opts, object_id):
 
     @csrf_protect_m
     def changeform_view(self, request, object_id=None, form_url="", extra_context=None):
-        with transaction.atomic(using=router.db_for_write(self.model)):
+        db = (
+            router.db_for_read(self.model)
+            if request.method in ("GET", "HEAD", "OPTIONS", "TRACE")
```

Comment:
There isn't currently a util for whether the request's method is read-only, so I've had to duplicate this set of methods a fair bit. Not sure this is the point we want to extract them into a util?